src/main/java/cinema/controllers/MovieSessionController.java

```diff
@@ -0,0 +1,46 @@
+package cinema.controllers;
+
+import cinema.model.MovieSession;
+import cinema.model.dto.request.MovieSessionRequestDto;
+import cinema.model.dto.response.MovieSessionResponseDto;
+import cinema.model.mapper.MovieSessionMapper;
+import cinema.service.MovieSessionService;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie_sessions")
+public class MovieSessionController {
+
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public void add(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.getMovieSession(movieSessionRequestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAll(@RequestParam (name = "movieId") Long id,
+                                                @RequestParam (name = "date") LocalDate showTime) {
+        List<MovieSession> movieSessions = movieSessionService.findAvailableSessions(id, showTime);
+        List<MovieSessionResponseDto> movieSessionsDto = new ArrayList<>();
```

Comment:
```suggestion
        List<MovieSessionResponseDto> movieSessionDtos = new ArrayList<>();
```
you are returning a list of DTOs